cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/template/etl/api/Emitter.java

```diff
@@ -32,4 +32,11 @@
    * @param value the object to emit
    */
   void emit(T value);
+
+  /**
+   * Emit an Error object.
+   *
+   * @param value the object to emit
+   */
+  void emitError(T value);
```

Comment:
don't thing this should emit the same type as normal emit.  Seems like it should be an object containing an error message, error code, and problem T.